soc/arm/st_stm32/common/soc_config.c

```diff
@@ -26,8 +26,10 @@
 static int st_stm32_common_config(const struct device *dev)
 {
 #ifdef CONFIG_LOG_BACKEND_SWO
+#if defined(DBGMCU_CR_TRACE_MODE)
 	/* TRACE pin assignment for asynchronous mode */
 	DBGMCU->CR &= ~DBGMCU_CR_TRACE_MODE_Msk;
+#endif
 	/* Enable the SWO pin */
 	DBGMCU->CR |= DBGMCU_CR_TRACE_IOEN;
```

Comment:
should TRACE_IOEN be set first or at the same time than TRACE_MODE=00
liek the `LL_DBGMCU_SetTracePinAssignment` function does
if TRACE_IOEN=0 when writing DBGMCU->CR &= ~DBGMCU_CR_TRACE_MODE_Msk; then TRACE_MODE=xx: TRACE pins not assigned --> useless
Only With TRACE_IOEN=1 is the TRACE_MODE written

Reply:
Maybe better to rewrite it using LL and explicitly mention SoC family:
```
#ifdef CONFIG_LOG_BACKEND_SWO
	/* Enable SWO trace asynchronous mode */
#if defined(CONFIG_SOC_SERIES_STM32WBX)
	LL_DBGMCU_EnableTraceClock();
#else
	LL_DBGMCU_SetTracePinAssignment(LL_DBGMCU_TRACE_ASYNCH);
#endif
#endif /* CONFIG_LOG_BACKEND_SWO */
```

Other series like H7 and L5 may have different enable procedure, but it looks like SWO is not supported for them by Zephyr by now.